components/financial_assistance/spec/controllers/financial_assistance/evidences_controller_spec.rb

```diff
@@ -0,0 +1,488 @@
+        before do
+          get :update_evidence, params: params_minus_update_reason
+        end
+
+        it 'sets an error flash message and redirects' do
+          expect(flash[:error]).to eq 'Please provide a verification reason.'
+          expect(response).to redirect_to main_app.verification_insured_families_path
+        end
+      end
+    end
+
+    describe 'POST #fdsh_hub_request' do
+      let(:request_double) { double }
+      context 'when the request determination is successful' do
+        before do
+          allow(Operations::Fdsh::RequestEvidenceDetermination).to receive(:new).and_return(request_double)
+          allow(request_double).to receive(:call).with(evidence).and_return(Success())
+          post :fdsh_hub_request, params: params
+        end
+
+        it 'sets a success flash message and redirects' do
+          expect(flash[:success]).to eq 'request submitted successfully'
+          expect(response).to redirect_to main_app.verification_insured_families_path
+        end
+      end
+
+      context 'when the request determination is not successful' do
+        before do
+          allow(Operations::Fdsh::RequestEvidenceDetermination).to receive(:new).and_return(request_double)
+          allow(request_double).to receive(:call).with(evidence).and_return(Failure())
+          post :fdsh_hub_request, params: params
+        end
+
+        it 'sets an error flash message and redirects' do
+          expect(flash[:error]).to eq 'unable to submit request'
+          expect(response).to redirect_to main_app.verification_insured_families_path
+        end
+      end
+    end
+
+    describe 'PUT #extend_due_date' do
+      before do
+        allow(FamilyMember).to receive(:find).and_return(family_member)
+        allow(family_member).to receive_message_chain(:family, :enrollments, :enrolled, :first).and_return(enrollment)
+      end
+
+      context 'when the evidence is type unverified and enrollment is present' do
+        it 'extends the due date and sets a success flash message' do
+          put :extend_due_date, params: params
+          expect(flash[:success]).to eq("#{evidence.title} verification due date was extended for 30 days.")
+        end
+      end
+
+      context 'when the evidence is not type unverified or enrollment is not present' do
+        it 'sets a danger flash message' do
+          put :extend_due_date, params: esi_params
+          expect(flash[:danger]).to eq("Applicant doesn't have active Enrollment to extend verification due date.")
+        end
+      end
+
+      # this scenario does not seem functionally possible; extend_due_on will return the result from building a new verification history
+      context 'when extending the due date fails' do
+        it 'sets a danger flash message' do
+          # put :extend_due_date, params: params
+          # expect(flash[:danger]).to eq("Unable to extend due date")
+        end
+      end
+    end
```

Comment:
Would like an opinion on this. The else block in the evidences_controller#extend_due_date does not look like it should ever really be hit, thus making this check unneeded.